pageserver/src/object_repository.rs

```diff
@@ -273,11 +273,13 @@ impl Timeline for ObjectTimeline {
         // Handle truncated SLRU segments.
         // XXX if this will turn out to be performance critical,
         // move this check out of the funciton.
-        if let RelishTag::Slru{slru: _slru, segno: _segno} = rel
+        if let RelishTag::Slru {
```

Comment:
Why do we need this special handling for SLRUs?

Reply:
Because otherwise we will return a stale or zeroed page for the truncated SLRU segment.

This is fine and sometimes even the desired behavior for relations (i.e. to reset VM and FSM forks),
but it doesn't work for SLRUs. I think we should refactor this function some more and explicitly specify "zero_if_not_found", but let's leave it as TODO for now.

Reply:
I've updated the comment. Hopefully, is more clear now.
Is there anything else I need to fix before merging?